src/onegov/org/forms/settings.py

```diff
@@ -1132,3 +1133,39 @@ class EventSettingsForm(Form):
         description=_('Enables website visitors to submit their own events'),
         default=True
     )
+
+
+def generate_timespans() -> (
+    tuple[tuple[str | datetime.timedelta, str], ...]
+):
+    return (
+        ('disabled', _('Disabled')),
+        (datetime.timedelta(days=180), _('6 months')),
+        (datetime.timedelta(days=365), _('1 year')),
+        (datetime.timedelta(days=365 * 2), _('2 years')),
+        (datetime.timedelta(days=365 * 3), _('3 years')),
+    )
```

Comment:
Why is this a function? As far as I can tell there's nothing dynamic being generated here.

I would move this into a global constant, e.g. `TIMESPAN_CHOICES`. 

Also it's worth noting that this doesn't actually work because both `SelectField` and `RadioField` default their `coerce` argument to `str`, i.e. all the values are turned into strings.

I think the sanest way to solve this would be to pass `coerce=int` and use integers for the choices, then you can use `0` for disabled. Then you can either override `process_obj`/`populate_obj` as you have or simplify things by leaving the setting as an integer and convert it to a timedelta in the cronjob.

Reply:
ah yes probably better than handling the strings. This itched me anyway, having to depend on string parsing of datetime for this critical part of the code.

Reply:
And  it's a function because we would have to duplicate it two times for the both fields of this form....

Reply:
you don't need a function to use the same choices in multiple fields. We have other fields that share common choices, such as `WEEKDAYS`, those are defined in a global variable as well.

Reply:
ah right, it only needs computed once.